src/main/java/seedu/address/logic/parser/AddressBookParser.java

```diff
@@ -128,46 +127,63 @@ private Command parseCountryCommands(String input) throws ParseException {
      * @throws ParseException if input does not conform to expected format
      */
     private Command parseClientCommands(String input) throws ParseException {
-        final Matcher secondaryMatcher = SECONDARY_COMMAND_FORMAT.matcher(input.trim());
-        if (!secondaryMatcher.matches()) {
+        final Matcher matcher = BASIC_COMMAND_FORMAT.matcher(input.trim());
+        if (!matcher.matches()) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
         }
 
-        String commandWord = secondaryMatcher.group("commandWord");
-        String arguments = secondaryMatcher.group("arguments");
-        commandWord = CLIENT_TYPE + " " + commandWord;
-        // todo: abstract away this parsing logic or use better regex matcher
-        StringTokenizer stringTokenizer = new StringTokenizer(arguments);
-        String nextWord = stringTokenizer.nextToken();
-        if (nextWord.equals("add")) {
-            // gather rest of the arguments:
-            StringBuilder stringBuilder = new StringBuilder();
-            while (stringTokenizer.hasMoreTokens()) {
-                stringBuilder.append(stringTokenizer.nextToken()).append(" ");
-            }
-
-            commandWord += (" " + nextWord);
-            arguments = " " + stringBuilder.toString();
+        final String commandType = matcher.group("commandType");
```

Comment:
Can rename this to clientCommandType? and the country one to countryCommandType?